central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -646,3 +646,89 @@ func (ds *datastoreImpl) RemovePLOPsWithoutProcessIndicatorOrProcessInfo(ctx con
 
 	return int64(len(plopsToDelete)), nil
 }
+
+// Removes PLOPs without poduids between a range of ids.
+func (ds *datastoreImpl) RemovePLOPsWithoutPodUIDOnePage(ctx context.Context, prevId string, nextId string) (int64, error) {
+	ds.mutex.Lock()
+	defer ds.mutex.Unlock()
+
+	query := fmt.Sprintf(deletePLOPsWithoutPoduidInPage, prevId, nextId)
+	commandTag, err := ds.pool.Exec(ctx, query)
+
+	if err != nil {
+		return 0, err
+	}
+
+	return commandTag.RowsAffected(), nil
+}
+
+// Given a set of rows with ids, returns the id of the last row. This is useful for pagination.
+func (ds *datastoreImpl) getLastIdFromRows(ctx context.Context, rows pgx.Rows) (string, error) {
+	id := ""
+
+	for rows.Next() {
+		if err := rows.Scan(&id); err != nil {
+			return "", pgutils.ErrNilIfNoRows(err)
+		}
+	}
+
+	return id, nil
```

Comment:
```suggestion
	return id, rows.Err()
```
Either do this here or the one below.  It just has to be done some place after rows.Next() is last called and before the defer of `rows.Close()` is done.

Reply:
I accepted the other suggestion instead of this one.